File: Your-Space-Backend/YourSpace.Data/Entities/EventGuest.cs
using YourSpace.Data.Enums;

namespace YourSpace.Data.Entities;

// Join row between Event and Person — hard-deletable (no DeletedAt). EventGuestStatus.Skipped
// already covers "deliberately excluded but still tracked"; a real delete here is for
// "added by mistake", so no soft-delete safety net is needed the way Person/Group/Event have one.
public class EventGuest
{
    public int Id { get; set; }

    public required int EventId { get; set; }
    public required int PersonId { get; set; }

    public EventGuestStatus Status { get; set; } = EventGuestStatus.NotInvited;

    // Chosen live, at the moment of actually inviting this person — never set at Person-creation time.
    public InviteMethod? InviteMethod { get; set; }

    public DateTime? InvitedAt { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public Event Event { get; set; } = null!;
    public Person Person { get; set; } = null!;
}

File: Your-Space-Mobile/lib/features/events/domain/entities/event_guest.dart
import 'package:equatable/equatable.dart';

import 'package:your_space_mobile/core/entities/invite_method.dart';

import 'event_guest_status.dart';

class EventGuest extends Equatable {
  final int id;
  final int eventId;
  final int personId;
  final String personName;
  final String? personPhoneNumber;
  final int groupId;
  final String groupName;
  final EventGuestStatus status;
  final InviteMethod? inviteMethod;
  final DateTime? invitedAt;

  const EventGuest({
    required this.id,
    required this.eventId,
    required this.personId,
    required this.personName,
    this.personPhoneNumber,
    required this.groupId,
    required this.groupName,
    this.status = EventGuestStatus.notInvited,
    this.inviteMethod,
    this.invitedAt,
  });

  @override
  List<Object?> get props => [
        id,
        eventId,
        personId,
        personName,
        personPhoneNumber,
        groupId,
        groupName,
        status,
        inviteMethod,
        invitedAt,
      ];
}
